starter-code/app.js

```diff
@@ -3,3 +3,40 @@ const app = express();
 const Chuck  = require('chucknorris-io');
 const client = new Chuck();
 
+app.set("views", __dirname + "/views");
+app.set("view engine", "ejs");
+
+app.get('/', (req, res) => {
+    res.render("index");
+  });
+
+app.get('/random', (req, res, next) => {
+    client
+    .getRandomJoke()
+    .then(response => {
+        res.send(`<p> ${response.value} </p>`);
+    })
+    .catch(err => {
+        console.log(err);
+    });
+})
+
+app.get('/categories', (req, res, next) => {
+    let cat;
+    
+    client
+    .getJokeCategories()
+    .then(response => {
+        cat = response;        
+    })
+    .catch(err => {
+        console.log(err);
+    });
+    res.render("categories", cat);
```

Comment:
CAUTION!!! You are rendering outside the 'then'. So your code will render the page before you have the response and cat will be empty (undefined).